File: src/main/java/com/example/progettoenterprise/dto/ItinerarioPreferitoDTO.java
package com.example.progettoenterprise.dto;
import com.example.progettoenterprise.data.entities.ItinerarioPreferito.Visibilita;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import jakarta.validation.constraints.*;
import java.time.LocalDate;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ItinerarioPreferitoDTO {

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private Long idItinerario;

    @NotBlank
    private String nome;

    private Boolean inCondivisione;

    private String proprietarioUsername;;

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private LocalDate dataCreazione;

    @NotNull
    private Visibilita visibilita;

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private List<ViaggioDTO> viaggiContenuti;
}

File: frontendMobile/app/src/main/java/com/example/enterprisemobile/model/ItinerarioPreferitoDTO.kt
package com.example.enterprisemobile.model

import com.google.gson.annotations.SerializedName

data class ItinerarioPreferitoDTO(
    @SerializedName("idItinerario") val idItinerario: Long? = null,
    @SerializedName("nome") val nome: String,
    @SerializedName("inCondivisione") val inCondivisione: Boolean? = null,
    @SerializedName("proprietarioUsername") val proprietarioUsername: String? = null,
    @SerializedName("dataCreazione") val dataCreazione: String? = null,
    @SerializedName("visibilita") val visibilita: Visibilita,
    @SerializedName("viaggiContenuti") val viaggiContenuti: List<ViaggioDTO>? = null
)